### DISCUSSION TITLE
Mac Os Big Sur Five Server not working

### DISCUSSION THREAD

**Discussion by iamk3v** [yandeu/five-server#3] [Help / Questions]
Hey since we moved here:

my fiveserver.config.js is as following:

```js
module.exports = {
    debugVSCode: true,
    browser: ["google chrome", "google-chrome", "google-chrome-stable", "chrome"],
  };
```

As far the the debug terminal goes, there's nothing popping up (see gif below)

https://gyazo.com/22ef204bbbc3fcd2f1eda275a78cd12c

**Comment by yandeu** [comment]
I have an old mac mini. I will try to install Big Sur on it.

**Reply by iamk3v** [reply]
Thank you! 

Although it could all be on my side.. But I'm clueless to as what I'm suppost to be looking for in order to fix this

**Reply by yandeu** [reply]
> Although it could all be on my side

It could, but I will check it.

---

But you could try to install the [`five-server npm package`](https://www.npmjs.com/package/five-server) to see if that works.

**Reply by iamk3v** [reply]
Update: 

Tried the NPM package, running five-server brings me to the root folder AND sends a 404 error in console:

<img width="671" alt="Screenshot 2021-04-06 at 14 19 50" src="https://user-images.githubusercontent.com/30670151/113709902-5a338300-96e3-11eb-8b78-1bdd763f8577.png">

Looks like its trying to start an index.html and not the HTML file I'm currently in?

Edit:

By the looks of it, it looks like that the five server is trying to find a index.html in my root folder instead of go into the folders where my actual file that I'm trying to open is..
I tried `five-server FILENAME`, but then its looking for `root/FILENAME/index.html`...?

**Reply by yandeu** [reply]
Unless you have a config file, five-server always looks relative to where it is executed from.

I have a `index.html` file on my Desktop (windows).

If I run `five-server index.html` inside `C:\Users\USERNAME\Desktop`, it will open that index file in the browser.

--- 

_Will soon be able to test it on mac. Installing updates..._

**Reply by iamk3v** [reply]
Okay a tiny update:

It works fine when i CD into my diagram folder (where the file is) and run `file-server FILENAME`.. But i would prefer to use the extension over node.. I loved the feature of the live server where you could just right click on the file and open it directly without having to cd into folders and run commands..

Could this be the case for the extension? that its looking for a index.html inside my root folder instead of actually being relative to where i right click and press Open with Five Server?

**Reply by yandeu** [reply]
I have made some updates and now have **High Sierra** where everything works perfectly fine.

I will continue updating...

**Reply by yandeu** [reply]
I have tested it with:
macOS Big Sur (11.2.3)
vscode (1.55.0)
chrome (89.0)
five-server extension (0.0.20)

Everything works perfectly.
_Chrome is my default browser._

**Reply by iamk3v** [reply]
Thats pretty interesting.. I have not touched a single piece of config or anything related to both live server and five server after I updated OS, yet I recieve these weird errors..

I suppose a reinstall of the extension won't do much? And as for "everything works perfectly" I take it as it works fine to right click and open with five server?

Super weird issue.. I can try to restart my Mac and reinstall extension, but I have a feeling that I will return with no success

**Reply by yandeu** [reply]
> I suppose a reinstall of the extension won't do much?

I don't think so.

> I take it as it works fine to right click and open with five server?

Yes, this works any anything else I've tested too.

---

Maybe update vscode? Have you set chrome as your default browser?

Maybe it is because of oneDrive? Have you tried to move your files?

**Reply by iamk3v** [reply]
Everything is updated to newest version and chrome is my macs default browser aswell as the config file you sent me to specify..

Onedrive hasnt Been an issue in the past, and it can read i guess as it opens the root folder when i just click on the Icon to go live. I will try though.. otherwise i guess ill just have to live with this.

Really do appreciate your effort to help me though!

**Reply by yandeu** [reply]
Just to be 100% sure. Does the following file exist?
`/Users/marius/OneDrive - Oslo Kommune Utdanningsetaten/IT/IT2_20_21/index.html`.
If it does, node.js can't access your files and it is probably the fault of oneDrive.

**Reply by iamk3v** [reply]
The file does not exist no. But it could be a onedrive issue? Even though my vsc should have access to it..

But then again the npm package works when i cd in to the folder, but not try to start from main folder.

Is there a way to see if vsc can access onedrive? Havent found anything on that

**Reply by yandeu** [reply]
Unfortunately, I don't know :/

**Reply by iamk3v** [reply]
> Unfortunately, I don't know :/

Worked fine with a non onedrive folder. Sorry to waste your time! but then its VSC and onedrive permissions that's messed up.. ill continue to search on where i could edit those perms (no luck for mac yet)
